src/debug-integration.adoc

```diff
@@ -11,6 +11,25 @@ When executing code due to an abstract command, the hart stays in debug mode
 and the rules outlined in Section 4.1 of cite:[riscv-debug-spec]
 apply.
 
+==== CHERI Mode when in Debug Mode
+
+When entering debug mode, the CHERI mode is always the same, and may be
+switchable depending on which extensions have been implemented.
+
+If {cheri_mode_ext_name} is implemented:
+
+* the core enters Legacy Mode when entering debug mode,
+ and the mode can be switched using <<MODESW>>.
```

Comment:
Why is this in this PR?  It seems unrelated to non-monotonic capability construction.

This will be nontrivial to implement for Rocket-type debuggers with a Debug ROM that contains instructions in a fixed ISA.

Reply:
because otherwise there's no way of switching to capability mode in debug mode. Previously MODESW was potentially illegal in debug mode.

Reply:
I continue be of the belief that, just as DXLEN is fixed at the maximum supported, so too should debug mode on a CHERI system always be capability mode rather than legacy mode

Reply:
Is it not reasonable on a legacy system that the debugger starts in legacy mode and then switches into cap mode? For software compatibility?

Reply:
Then you make legacy the primary mode, which the spec is //not// doing in other places.

Reply:
(and everything that wants to use capability mode, e.g. to read a capability from a CSR, has to waste precious instruction buffer space switching to capability mode)

Reply:
cap mode is the default now, and MODESW is optionally supported